packages/salesforcedx-vscode-replay-debugger/src/core/logContext.ts

```diff
@@ -167,7 +168,7 @@ export class LogContext {
     const processedSignature = signature.endsWith(')')
       ? signature.substring(0, signature.lastIndexOf('.'))
       : signature;
-    const typerefMapping = this.session.getBreakpointUtil().getTyperefMapping();
+    const typerefMapping = BreakpointUtil.getInstance().getTyperefMapping();
```

Comment:
What's the reasoning behind changing the call from using `this.session.getBreakpointUtil()` to `BreakpointUtil.getInstance()`.

Reply:
Nick, I changed the class to be a singleton class. Previously, this class was only used in the adapter and the instance was stored on the apedReplayDebug. With the addition of processing breakpoints in the extension it was needed in both processes and had to be setup in each. In the extension code it's loaded on activation and then used in the breakpoint event, creating the singleton was better, cleaner way to do this. Being a singleton also has the added bonus of making it easier to reload once we get any filesystem changed events hooked up.

Reply:
Yup the singleton makes sense. We usually create a Singleton and just export that instance out. See https://github.com/forcedotcom/salesforcedx-vscode/blob/develop/packages/salesforcedx-vscode-core/src/channels/index.ts#L9

We don't have classes calling `.getInstance()` directly.

Reply:
Sure. Done.